src/include/network/services/host.rb

```diff
@@ -249,15 +249,14 @@ def HostsMainDialog(standalone)
           UI.ChangeWidget(Id(:table), :Items, table_items)
           modified = true
         elsif ret == :next
-          # check_
           next if !modified
           Host.clear
 
           table_items.each do |row|
-            encoded_aliases = Punycode.EncodePunycodes([row.fetch(3, "")])
-            encoded_canonical = Punycode.EncodeDomainName(row.fetch(2, ""))
+            encoded_aliases = Punycode.EncodePunycodes([row[3] || ""])
```

Comment:
Change seems to make sense, but I really miss a test. Could you please add a simple one?

BTW, if `Punycode.EncodeXYZ` accepts `nil`, you also could write the code like

```ruby
table_items.each do |key, canonical, alias|
  .  .  .
end
```

or even avoid the `nil` values

```ruby
table_items.each do |key = '', canonical = '', alias = ''|
  .  .  . 
end
```

Reply:
well adding simple test  is not that simple here ... the method is spread over 250 lines and is something like all-in-one dialog handler.